### PR TITLE
feat: Add Deploy Command and Update Configurations for Compatibility

### PR DESCRIPTION
**Pull request by epsjunior**
## Overview

This PR introduces a new `genlayer deploy` command to the CLI, enabling users to deploy smart contracts to the blockchain with ease. Several related changes were made to ensure compatibility with dependencies and the latest versions of tools.

---

### `genlayer deploy` Command

The new command allows users to deploy intelligent contracts

#### Options:

- `--contract <contractPath>`: Path to the intelligent contract file to deploy.
- `--args <args...>`: Positional arguments for the contract (space-separated values).

#### Usage Examples:

1. **Deploy a contract:**
   ```bash
   genlayer deploy --contract ./path/to/contract
   ```

2. **Deploy with positional arguments:**
   ```bash
   genlayer deploy --contract ./path/to/contract --args 1 2 3
   ```

### 1. Switch to Bundler for `tsconfig.json`
- Updated `module` and `moduleResolution` to `bundler` for compatibility with the `genlayer-js` library.

### 2. `esbuild` Configuration
- Updated to use `es2020` to support `viem` library requirements.

### 3. Hardhat Service Integration
- Configured Hardhat service to fetch from Docker registry.
- Updated all environment variables to ensure compatibility with the latest version of Studio.

---

### End-to-End Tests
- Tested the `genlayer deploy` command with the latest version of Studio using both `Ollama` and `OpenAI` validators.
- Verified contract deployment and interaction using the both front end and terminal to confirm retro compatibility.

### Deployment Command Tests
- Successfully tested deploying the `user_storage.gpy`

### CODE REVIEW

**Review by cristiam86 — CHANGES_REQUESTED** [review]

File: src/commands/general/deploy.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
+import fs from "fs";
+import { createClient, createAccount } from "genlayer-js";
+import { simulator } from "genlayer-js/chains";
+import type { GenLayerClient } from "genlayer-js/types";
+import { getPrivateKey } from "../../lib/utils/getPrivateKey";
+
+export interface DeployOptions {
+  contract?: string;
```
[the remark this answers is not in the document]
Comment (epsjunior) [reply src/commands/general/deploy.ts:8]: yes, I considered using requiredOption for --contract, so Commander would throw an error if the user doesn’t provide this option. However, I plan to add a feature that allows running a script to deploy contracts, making the --contract option optional. The if statement is a temporary solution

File: src/commands/general/deploy.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
+import fs from "fs";
```
[the remark this answers is not in the document]
Comment (epsjunior) [reply src/commands/general/deploy.ts:1]: I initially thought that all commands without subcommands would go into this folder, but you’re right, it’s better to organize them into a separate "contracts" folder :)

[the remark this answers is not in the document]
Comment (epsjunior) [reply src/commands/general/deploy.ts]: fixed on https://github.com/yeagerai/genlayer-cli/pull/153/commits/e548ced633272150476bb18d20549fc2a04214b1

File: src/commands/general/index.ts [hunk 3]
```diff
@@ -23,5 +23,17 @@ export function initializeGeneralCommands(program: Command) {
     .option("--reset-db", "Reset Database", false)
     .action((options: StartActionOptions) => startAction(options, simulatorService));
 
+  program
```
[the remark this answers is not in the document]
Comment (epsjunior) [reply src/commands/general/index.ts]: fixed on https://github.com/yeagerai/genlayer-cli/pull/153/commits/e548ced633272150476bb18d20549fc2a04214b1

**Review by mpaya5 — APPROVED** [review]

**Pull request merged**